features/bsm/service/impl/src/main/java/org/opennms/netmgt/bsm/service/internal/DefaultBusinessServiceStateMachine.java

```diff
@@ -356,4 +364,14 @@ public String transform(GraphEdge edge) {
             m_rwLock.readLock().unlock();
         }
     }
+
+    @Override
+    public BusinessServiceGraph getGraph() {
+        m_rwLock.readLock().lock();
+        try {
+            return m_g;
```

Comment:
Can we return a copy of the graph instead of a reference? There's also no point in using a read lock when returning the reference. I'm mostly worried that callers may alter the graph in some way.